### ISSUE TITLE
when quit gvim  throw Exception

### ISSUE THREAD

**Issue by foxundermoon** [Shougo/vimproc.vim#202]
```
Error detected while processing function <SNR>188_finalize..<SNR>..188_vp_dlclose..<SNR>188_libcall: 
E605: Exception not caught : vimproc : vp_dlclose:['vp_stack_pop_mum:sscanf error']
E171: Missing :endif
```

above this is error message. what's wrong about me. 

this is the function throw exception,but i don't know why it throw?
```vimL
function! s:libcall(func, args) "{{{
  " End Of Value
  let EOV = "\xFF"
  let args = empty(a:args) ? '' : (join(reverse(copy(a:args)), EOV) . EOV)
  let stack_buf = libcall(g:vimproc#dll_path, a:func, args)
  let result = s:split(stack_buf, EOV)
  if get(result, -1, 'error') != ''
    if stack_buf[len(stack_buf) - 1] ==# EOV
      " Note: If &encoding equals "cp932" and output ends multibyte first byte,
      "       will fail split.
      return result
    endif
    let s:lasterr = result
    let msg = vimproc#util#iconv(string(result),
          \ vimproc#util#systemencoding(), &encoding)

    throw printf('vimproc: %s: %s', a:func, msg)
  endif
  return result[:-2]
endfunction"}}}
```

**Comment by Shougo** [comment]
It seems the handle is invalid.
But I don't know why.

**Comment by ukatama** [comment]
I had the same problem.

I tested on Windows 10 x64 / Visual Studio 2015 / Vim 7.4.

Exception would had been thrown from [line 91 of autload/vimstack.c](https://github.com/Shougo/vimproc.vim/blob/master/autoload/vimstack.c#L191).

```c
    if (sscanf(top, fmtbuf, ptr, &n) != 1 || top[n] != VP_EOV)
        return "vp_stack_pop_num: sscanf error";
```

If the variable `top` is string `"3\xFF\0"` and `fmtbuf` is string `"%d%n\0"`, the variable `n` should be 1. But it was 2.

I made changes to the `top[n]` to `top[n-1]`, the exception no longer occurs.
However, I have not tried in the other environment.

**Comment by foxundermoon** [comment]
my system also is win10 x64,build by Visual Studio 2015
may be it  can not run well on the new  operation system

**Comment by Shougo** [comment]
@k-takata Can you test the problem?

**Comment by k-takata** [comment]
I tested this program: https://gist.github.com/k-takata/07764e23815a3959d431
Only VS2015 sets different value to `n`. (See a comment in the gist.)
I'm not sure this is a bug of VS2015 or intended change.

**Comment by Shougo** [comment]
Oh, it seems bug...
vimproc can handle it?

**Comment by foxundermoon** [comment]
it's visual studio's bug?  or os
i;m try to clone the new source and rebuld  but error also display.
use the old  dll build by vs2013 @ win8.1 also error.
how to fix it.

**Comment by Shougo** [comment]
It seems VS2015 only problem.

> i;m try to clone the new source and rebuld but error also display.
> use the old dll build by vs2013 @ win8.1 also error.

This is same error message?

@k-takata Can you test the problem in VS2013 @ Windows 8.1?

**Comment by Shougo** [comment]
http://blogs.msdn.com/b/vcblog/archive/2014/06/03/visual-studio-14-ctp.aspx

scanf is changed in Visual Studio 2015.

**Comment by Shougo** [comment]
@foxundermoon You can try another vimproc build in kaoriya Vim.

http://files.kaoriya.net/goto/vim74w64

**Comment by k-takata** [comment]
> @k-takata Can you test the problem in VS2013 @ Windows 8.1?

I checked the [test program](https://gist.github.com/k-takata/07764e23815a3959d431) with VS2013 on Win7. It worked as expected. (I don't use Win8.)

**Comment by Shougo** [comment]
This is Visual Studio's bug.
But we must fix it...
I will create the branch for it.

**Comment by Shougo** [comment]
I created the branch for it.

**Comment by Shougo** [comment]
Fixed by #206.
